src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/EntityAIMeleeGuard.java

```diff
@@ -129,8 +133,28 @@ protected AIState huntDown()
             targetEntity = this.worker.getLastAttacker();
         }
 
-        if (!targetEntity.isEntityAlive() || checkForToolOrWeapon(ToolType.SWORD))
+        final AbstractEntityBarbarian barbarian = BarbarianUtils.getClosestBarbarianToEntity(worker, 20D);
+
+        if (barbarian != null)
+        {
+            targetEntity = barbarian;
+        }
+
+        if (targetEntity != null && worker.getColony() != null)
+        {
+            final List<EntityLivingBase> targets = worker.getColony().getGuardTargets();
+            if (targets.stream().noneMatch(entity -> entity == targetEntity))
+            {
+                targets.add(targetEntity);
+            }
+            worker.getColony().setGuardTargets(targets);
+        }
+
+        if (targetEntity != null && !targetEntity.isEntityAlive() || checkForToolOrWeapon(ToolType.SWORD))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add parentheses to make the operator precedence explicit. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS864)